### DISCUSSION TITLE
Usual CPU usage?

### DISCUSSION THREAD

**Discussion by KDYR** [wmbusmeters/wmbusmeters#841] [Q&A] [answered]
What is the typical (or reasonable) CPU consumption for the WMBusMeters HA add-on? After installing the add-on my CPU usage has risen significantly: From around 6% to constantly above 22% (Rpi4B). CPU temperature has also risen by 10°C.
![image](https://user-images.githubusercontent.com/16138808/216761325-f9e5383b-9655-4766-839d-4d29369a05f1.png)
Glanses reports this:
![image](https://user-images.githubusercontent.com/16138808/216761510-25b4c136-8703-4ff4-8b58-fc3652f58e60.png)
...and this:
![image](https://user-images.githubusercontent.com/16138808/216761638-eec272d2-5643-4b44-a3c2-7ef2653d7637.png)
Is this the CPU usage others see too? Is there any optimisation possible?

**Comment by KDYR** [comment]
Forgot to mention:
At this stage I am using the SDR dongle and wmbusmeters to monitor a single Kamstrup FlowIQ 2200, and this is my HA configuration:
```
data_path: /config/wmbusmeters
enable_mqtt_discovery: true
conf: |-
  loglevel=normal
  device=auto:t1
  donotprobe=/dev/ttyAMA0
  logtelegrams=false
  format=json
  logfile=/dev/stdout
  shell=/wmbusmeters/mosquitto_pub.sh "wmbusmeters/$METER_NAME" "$METER_JSON"
meters:
  - |-
    name=WaterMeter
    driver=flowiq2200
    id=XXXX...
    key=XXXXXXXXXXXXXXX...
mqtt: {}

```

**Comment by weetmuts** [comment, marked as answer]
The rtl_wmubs radio decoding just takes a lot of cpu. Check the README for how to tune the settings, if you do not need to listen to c1,t1 and s1 at the same time etc.

You can also buy an im871a which does the radio decoding in hardware. The wmbusmeters telegram decoding takes minimal cpu time.

**Reply by KDYR** [reply]
THX!
Tried to read the README, but didn't immediately 'decode' all the - for me - special information. Will have to study harder, it seems. At this stage I do not even know which telegram-type the data from my Kamstrup FlowIQ 2200 is based on. So, I got a working setup - but don't really know why. 
Any other hints than to investigate if I can disable some of the c1,t1 and s1?
Thx again.

**Reply by KDYR** [reply]
Btw, it seems the im871a is sold out all over :(

**Comment by weetmuts** [comment]
it is c1

**Reply by KDYR** [reply]
...but why is my setup with device=auto:t1 working? Isn't this saying "probe all ports" + "decode only t1"

**Reply by weetmuts** [reply]
the rtl wmbus reports both c1 and t1 whether you want it or not! :-) 

also the setting of the link mode in wmbusmeters is not carried to rtl wmbus. you have to modify the command line to rtl wmbus. 

something to cleanup in the future if i have the time.
